### DISCUSSION TITLE
Do digital products still require a warehouse and stock level?

### DISCUSSION THREAD

**Discussion by Cameron485** [saleor/saleor#10794] [Q&A]
I am currently evaluating Saleor and have been using the excellent docs and have a local instance running with docker compose - works a treat! I have created a non shippable digital product type and the associated variants with track invetory false. I have no warehouses added as it's a digital only product. When trying to create a checkout I receive an `INSUFFICIENT_STOCK` error. Is it mandatory to have a warehouse if you are only planning on having digital products?

Saleor 3.7.0-a

**Comment by karolkielecki** [comment]
Hi @Cameron485, thanks for asking. Yes--you need to create warehouses for digital products. Currently, we have other priorities, but it's on our radar. We'll keep you updated. 

Cheers,

**Comment by karolkielecki** [comment]
By the way, you can follow the progress on that issue here: https://github.com/saleor/saleor/issues/7531#issuecomment-1268456946. We've just introduced an idea of how to manage digital products that don't need warehouses.
